conda_smithy/lint_recipe.py

```diff
@@ -515,6 +515,18 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
                 " form. See lines %s." % (bad_lines,)
             )
 
+    # 25: require a lower bound on python version
+    if build_section.get("noarch") == "python" and not outputs_section:
+        for req in run_reqs:
+            if req.startswith("python") and req != "python":
```

Comment:
This should make it work for things like `python-foo` as well

```suggestion
            if (req.strip().split()[0] == "python") and (req != "python"):
```